hazelcast/src/main/java/com/hazelcast/jet/impl/execution/ExecutionContext.java

```diff
@@ -291,7 +291,7 @@ public boolean terminateExecution(@Nullable TerminationMode mode, Throwable caus
 
         synchronized (executionLock) {
             if (mode == null) {
-                cancellationFuture.completeExceptionally(cause);
+                cancellationFuture.completeExceptionally(new ExecutionCancellationException(cause));
```

Comment:
Maybe propagate the message so the cause is on the first line of the stacktrace, not somewhere in the middle.